scalameter-core/src/main/scala/org/scalameter/Measurer.scala

```diff
@@ -407,6 +418,47 @@ object Measurer {
     def units = "#"
   }
 
+  private type Primitive = Boolean with Char with Byte with Short with Int with Long with Float with Double
+
+  /** Counts autoboxed by a Scala compiler values.
+   *
+   *  @param primitives primitive types whose autoboxing will be counted.
+   *                    Note that leaving it empty means that autoboxing of all primitive types will be counted.
+   */
+  class AutoBoxingCounting(primitives: Class[_ >: Primitive]*) extends InvocationCounting {
+    val matcher: InvocationCountMatcher = {
+      import InvocationCountMatcher._
+
+      val primitiveToBoxedMethod: Map[Class[_ >: Primitive], String] = Map(
+        classOf[Boolean] -> "boxToBoolean",
+        classOf[Char] -> "boxToCharacter",
+        classOf[Byte] -> "boxToByte",
+        classOf[Short] -> "boxToShort",
+        classOf[Int] -> "boxToInteger",
+        classOf[Long] -> "boxToLong",
+        classOf[Float] -> "boxToFloat",
+        classOf[Double] -> "boxToDouble"
+      )
+
+      InvocationCountMatcher(
+        classMatcher = ClassMatcher.ClassName(classOf[BoxesRunTime]),
+        methodMatcher = MethodMatcher.Regex((
+          if (primitives.nonEmpty)
+            new Regex(primitives.map(p => s"(${primitiveToBoxedMethod(p)})").mkString("^", "|", "$"))
+          else
+            "^boxTo\\w+$".r
```

Comment:
This is actually surprising and somewhat unexpected.
I'd suggest to instead have a factory method that creates the `BoxingCount` for all the primitive types.